bitcoin/BRWalletManager.c

```diff
@@ -710,18 +710,54 @@ BRWalletManagerScan (BRWalletManager manager) {
                                             });
 }
 
+extern BRTransaction *
+BRWalletManagerCreateTransaction (BRWalletManager manager,
+                                  BRWallet *wallet,
+                                  uint64_t amount,
+                                  const char *addr) {
+    BRTransaction *transaction = BRWalletCreateTransaction (wallet, amount, addr);
+    if (NULL != transaction) {
+        assert (NULL != manager->client.funcTransactionEvent);
+        manager->client.funcTransactionEvent (manager->client.context,
+                                              manager,
+                                              manager->wallet,
+                                              transaction,
+                                              (BRTransactionEvent) {
+                                                  BITCOIN_TRANSACTION_CREATED
+                                              });
+    }
+
+    return transaction;
+}
+
+extern int
+BRWalletManagerSignTransaction (BRWalletManager manager,
+                                BRTransaction *transaction,
+                                const void *seed,
+                                size_t seedLen) {
+    int r = (1 == BRWalletSignTransaction (manager->wallet, transaction, seed, seedLen) ? 1 : 0);
+    if (r) {
+        assert (NULL != manager->client.funcTransactionEvent);
+        manager->client.funcTransactionEvent (manager->client.context,
+                                              manager,
+                                              manager->wallet,
+                                              transaction,
+                                              (BRTransactionEvent) {
+                                                  BITCOIN_TRANSACTION_SIGNED
+                                              });
+    }
+
+    return r;
+}
+
```

Comment:
I don't see a problem with the above functions as they are consistent w/ current usage.  But recall we've discussed holding a signed transaction if the transaction can be changed subsequently.  (The change invalidates the signature - seen on ETH transactions).  And how maybe create+sign+submit need to be atomic.

Reply:
Great!